src/V3ThreadPool.cpp

```diff
@@ -51,9 +52,21 @@ void V3ThreadPool::enqueue(std::function<void()>&& f) {
 }
 
 void V3ThreadPool::wait() {
-    while (m_pendingJobs.load(std::memory_order_acquire) > 0 && !m_shutdown) {
-        std::this_thread::yield();
+    // Spin briefly first, as jobs often complete quickly
+    for (int i = 0; i < VL_LOCK_SPINS; ++i) {
+        if (m_pendingJobs.load(std::memory_order_acquire) == 0) break;
+        if (m_shutdown.load(std::memory_order_acquire)) break;
+        VL_CPU_RELAX();
     }
+    // Fall back to condition variable wait to avoid wasting CPU cycles.
+    // Skip if no pending jobs or shutting down (checked above in spin loop).
+    if (m_pendingJobs.load(std::memory_order_acquire) > 0 && !m_shutdown) {
+        V3LockGuard lock{m_mutex};
+        while (m_pendingJobs.load(std::memory_order_acquire) > 0 && !m_shutdown) {
+            m_completionCV.wait(m_mutex);
+        }
+    }
```

Comment:
Please use the lambda form of condition variable `wait`, to be consistent with the existing code.